src/main/java/org/owasp/html/HtmlEntities.java

```diff
@@ -28,13 +28,2129 @@
 
 package org.owasp.html;
 
+import java.util.Map;
+
 import com.google.common.collect.ImmutableMap;
 
 /**
  * Utilities for decoding HTML entities, e.g., {@code &amp;}.
  */
 final class HtmlEntities {
 
+  /** A trie that maps entity names to codepoints. */
+  public static final Trie ENTITY_TRIE;
+
+  private static final int LONGEST_ENTITY_NAME;
+
+  static {
+    final ImmutableMap.Builder<String, Integer> builder = ImmutableMap.builder();
+
+    // Source data: https://dev.w3.org/html5/html-author/charref
```

Comment:
Why prefer this to the relevant section of the HTML spec? https://html.spec.whatwg.org/multipage/named-characters.html#named-character-references

Reply:
No reason. It was the source I found. I will update to use the linked spec. :+1:

Reply:
Updated. Evidently some named references refer to 2 code points. Since all such references only refer to BMP code points, we can still pack those 2 code points into a 32-bit integer and not have to change the signature of `decodeEntityAt`, but I had to add some extra bit shifting.